Content.Server/CriminalRecords/Systems/GeneralCriminalRecordConsoleSystem.cs

```diff
@@ -0,0 +1,289 @@
+{
+    [Dependency] private readonly UserInterfaceSystem _userInterface = default!;
+    [Dependency] private readonly StationSystem _stationSystem = default!;
+    [Dependency] private readonly StationRecordsSystem _stationRecordsSystem = default!;
+    [Dependency] private readonly RadioSystem _radioSystem = default!;
+    [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+    [Dependency] private readonly AccessReaderSystem _accessReaderSystem = default!;
+    [Dependency] private readonly PopupSystem _popupSystem = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, BoundUIOpenedEvent>(UpdateUserInterface);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, SelectGeneralCriminalRecord>(OnKeySelected);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, GeneralStationRecordsFilterMsg>(OnFiltersChanged);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, RecordModifiedEvent>(UpdateUserInterface);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, AfterGeneralRecordCreatedEvent>(UpdateUserInterface);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, CriminalRecordArrestButtonPressed>(OnButtonPressed);
+        SubscribeLocalEvent<GeneralCriminalRecordConsoleComponent, CriminalStatusOptionButtonSelected>(OnStatusSelected);
+    }
+
+    private void OnFiltersChanged(EntityUid uid,
+        GeneralCriminalRecordConsoleComponent component, GeneralStationRecordsFilterMsg msg)
+    {
+        if (component.Filter == null ||
+            component.Filter.Type != msg.Type || component.Filter.Value != msg.Value)
+        {
+            component.Filter = new GeneralStationRecordsFilter(msg.Type, msg.Value);
+            UpdateUserInterface(uid, component);
+        }
+    }
+
+    private void UpdateUserInterface<T>(EntityUid uid, GeneralCriminalRecordConsoleComponent component, T ev)
+    {
+        UpdateUserInterface(uid, component);
+    }
+
+    private void OnKeySelected(EntityUid uid, GeneralCriminalRecordConsoleComponent component,
+        SelectGeneralCriminalRecord msg)
+    {
+        component.ActiveKey = msg.SelectedKey;
+        UpdateUserInterface(uid, component);
+    }
+
+    private void OnButtonPressed(EntityUid uid, GeneralCriminalRecordConsoleComponent component,
+        CriminalRecordArrestButtonPressed msg)
+    {
+        if (msg.Session.AttachedEntity is not {Valid: true} mob) return;
+        if (!CanUse(mob, uid))
+        {
+            _popupSystem.PopupEntity(Loc.GetString("general-criminal-record-permission-denied"), uid, msg.Session);
+            return;
+        }
+
+        TryComp<SecurityInfoComponent>(msg.Session.AttachedEntity, out var secInfo);
+
+        if (msg.Reason != string.Empty && msg.Name != null && msg.Session.AttachedEntity != null && secInfo != null)
```

Comment:
this looks like duplicated code - i think you could generalize some of this into multiple methods if possible